codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsRuntimeConfig.java

```diff
@@ -88,6 +93,8 @@ public void addConfigInterfaceFields(
                 .write("credentialDefaultProvider?: (input: any) => __Provider<__Credentials>;\n");
         writer.writeDocs("Provider function that return promise of a region string")
                 .write("regionDefaultProvider?: (input: any) => __Provider<string>;\n");
+        writer.writeDocs("Provider function that return promise of a maxAttempts string")
+                .write("maxAttemptsDefaultProvider?: (input: any) => __Provider<string>;\n");
```

Comment:
Why is input any?

Reply:
I'm not sure why input has been `any` for all providers historically.
It could be because of differences between browser and Node.js implementations.